pandas/tools/tests/test_merge_asof.py

```diff
@@ -221,6 +221,117 @@ def test_missing_right_by(self):
         expected.loc[expected.ticker == 'MSFT', ['bid', 'ask']] = np.nan
         assert_frame_equal(result, expected)
 
+    def test_multiby(self):
```

Comment:
Good idea. I've added it to `test_on_specialized_type()` and `test_on_specialized_type_by_int()`.